main.ts

```diff
@@ -122,6 +122,23 @@ class Translator {
     return true;
   }
 
+  visitNamedParameter(paramDecl: ts.ParameterDeclaration) {
+    this.visitEachIfPresent(paramDecl.decorators);
+    if (paramDecl.type) {
+      // TODO(martinprobst): These are currently silently ignored.
+      // this.reportError(paramDecl.type, 'types on named parameters are unsupported');
+    }
+    this.visit(paramDecl.name);
+    if (paramDecl.initializer) {
+      if (paramDecl.initializer.kind !== ts.SyntaxKind.ObjectLiteralExpression) {
+        this.reportError(paramDecl, 'named parameter initializer must be empty object literal');
+      }
+      if ((<ts.ObjectLiteralExpression>paramDecl.initializer).properties.length > 0) {
+        this.reportError(paramDecl, 'non-empty named parameter initializers are unsupported');
```

Comment:
I'd say the same as above, `initializers for named parameters must be empty object literals`, fewer negations and tells you what to do.